### ISSUE TITLE
Feature: Simplify the domain example and refactor

### ISSUE THREAD

**Issue by peterschutt** [starlite-api/starlite-pg-redis-docker#76]
```
    app
    | main.py
    |
    └───controllers
    |
    └───domain
    |
    └───lib
    |
    └───scripts
    |
    └───workers
    ```

This is more like the structure that I've settled on - nicer organization than the current app.

The main components are:

- main.py - instantiates the `Starlite` application and uses it for orchestration of db, cache, task queue
- controllers - where all the routes are defined, structured as directory or module per domain with directories matching path nesting
- domain - defines models, service, and repository objects allowing them to be shared between controllers and workers
- lib - where all the stuff goes that doesn't really change from one app to the next - I've factored this out to a separate library but we should have it included here
- scripts - standalone scripts configured as `tool.poetry.scripts`
- workers - async job queue workers (#65)

I also think we should simplify the domain down to something easier to grok, `Books` and `Authors`?

Suggestions welcome!

**Comment by florimondmanca** [comment]
Hi there

> I also think we should simplify the domain down to something easier to grok, Books and Authors?

I navigated this example repo as I'm curious of adopting Starlite in a "Clean Architecture" setup, so the core/domain structure appealed to me. But I agree with this -- I wasn't able to understand really what the example domain was about. It seems a bit too meta.

Something like a book library management system with `Book`, `Author`, perhaps `Tag` (for illustrating M2M relationships), could be easier to understand, indeed.

**Comment by peterschutt** [comment]
Thanks for the input @florimondmanca - the current domain is the very early stages of a very small slice of an esoteric application that I've been building at work.

**Comment by cofin** [comment]
I think it would be good to include a User model as well.  I can appreciate that many won't be doing authentication from the service, but I think it helps to show the power of the framework.

**Comment by peterschutt** [comment]
Good point @cofin. Will work that in for sure.

**Comment by peterschutt** [comment]
So I'm going to make it all based around the SQLAlchemy models.

Initial pattern had Service object dealing with pydantic models, and repository objects dealing with sqlalchemy models and I never liked the duplication and crossover of roles.

All this sqlalchemy plugin work should make it nice and easy to be dealing with the sqlalchemy models right to the controller edge.

Two of the reasons I had both pydantic and sqlalchemy models:

1. nice to know that operations inside the service object cannot accidentally modify db state
2. sqlalchemy typing was never that great

But I can address 1. by always expunging the instances from the session before they are returned from the repository, and now that sqlalchemy has better typing support 2. is a bit of a non-problem as well.

So, service object and repository object always only deal with sqlalchemy models, and autogenerated pydantic models for validation/serialization/documentation generation - hopefully using the types annotations of the sqlalchemy models to generate the pydantic models.

**Comment by ReznikovRoman** [comment]
Hi @peterschutt ! Thanks for the awesome framework and in particular for this comprehensive example!

What do you think about the [following structure](https://github.com/ets-labs/newsfeed)? So you still have `domain` and `lib`, but you can also add another component - `infrastructure`. For example, you can put there your [Worker](https://github.com/starlite-api/starlite-pg-redis-docker/blob/e62c04713996d6129c9e96f488645cb47f088508/app/lib/worker.py#L41), [Redis](https://github.com/starlite-api/starlite-pg-redis-docker/blob/e62c04713996d6129c9e96f488645cb47f088508/app/lib/redis.py#L7) and probably [SQLAlchemy related](https://github.com/starlite-api/starlite-pg-redis-docker/blob/e62c04713996d6129c9e96f488645cb47f088508/app/lib/sqlalchemy_plugin.py#L35) stuff.

I also think it would be greate to add an example of API versioning (e.g., placing your controllers inside of `app.api.v1.controllers` package or something like that; and slightly updating a `router`). How do you feel about that?

**Comment by peterschutt** [comment]
Hey @ReznikovRoman - thanks for your kind words and interest in the project.

I have added the `/v1` prefix to the router as you suggest. I'm not too fussed either way on this as if ever needing a `/v2` one can simply add the prefix then - but I like to beat the drum about consistency so I suppose I should be consistent about being consistent. I haven't altered the project structure at this stage however, as the domain example is still quite simple.

WRT your suggestion about having an `infrastructure` directory, I'm certainly not opposed to making the structure within `lib` a little less flat. The idea behind the `lib` folder is that it is the configuration that one would typically carry from one project to the next, so something like `lib/infrastructure` would be fine by me.

On this last point, have a look at [starlite-saqpg](https://github.com/topsport-com-au/starlite-saqpg/) as an example of such. It is not finished - and needs to be brought into line with the improvements that I've worked into this example, but it is basically what `lib` is in this project, but intended to be portable. The main thing is the [plugin](https://github.com/topsport-com-au/starlite-saqpg/blob/main/src/starlite_saqpg/init_plugin.py) which would be used like `app = Starlite(on_app_init=[starlite_saqpg.ConfigureApp()])`. Plenty of work still to be done on that project, but I'll be turning my attention to it shortly.

Also, please have a look at the major overhaul of this app that I've just merged which pretty much implements the structure that I've set out above.

**Comment by peterschutt** [comment]
Closed in #94
